src/tsolvers/lasolver/LABounds.cc

```diff
@@ -94,21 +93,21 @@ void LABoundStore::ensureReadyFor(LVRef v) {
     }
 }
 
-void LABoundStore::updateBound(BoundInfo bi) {
-    auto & varBounds = getBounds(bi.v);
-    for (LABoundRef bound : {bi.ub, bi.lb}) {
-        unsigned idx = varBounds.size();
-        varBounds.push(bound);
+void LABoundStore::placeLastTwoBounds(LVRef v) {
+    auto & varBounds = getBounds(v);
+    assert(varBounds.size_() >= 2);
+    unsigned indexToSort = varBounds.size_() - 2;
+    for (unsigned index : {indexToSort, indexToSort + 1}) {
+        LABoundRef bound = varBounds[index];
+        (void)bound;
         bound_lessthan lessthan(ba);
-        for (; idx > 0; --idx) {
-            if (lessthan(varBounds[idx - 1], varBounds[idx])) {
-                break;
-            }
-            std::swap(varBounds[idx - 1], varBounds[idx]);
-            ba[varBounds[idx]].setIdx(LABound::BLIdx{idx});
+        while (index > 0 and not lessthan(varBounds[index - 1], varBounds[index])) {
```

Comment:
```suggestion
        for (; index > 0 and not lessthan(varBounds[index - 1], varBounds[index]); --index) {
```
Not too important but I would still use `for` since we use the loop variable in a standard way.

Reply:
I prefer `while` variant. Seems more readable to me, but that's personal taste.
I can change it if you insist.

Reply:
I do not.